scripts/Engineering/EnginX.py

```diff
@@ -622,28 +622,41 @@ def _save_out(run_number, focus_directory, focus_general, output, join_string, b
 
     """
     # work out where to save the files
-    filename = os.path.join(focus_directory, join_string.format(run_number, bank_id))
-    hdf5_name = os.path.join(focus_directory, run_number + ".hdf5")
+    dat_name, genie_filename, gss_name, hdf5_name, nxs_name = _find_focus_file_location(bank_id, focus_directory,
```

Comment:
I think a better name for the `join_string` could be something like `enginx_file_name_format`